src/test/app/SetTrust_test.cpp

```diff
@@ -17,50 +17,190 @@
-        env(pay(env.master, alice, STAmount{ threelineReserve - baseReserve }));
-        env(trust(alice, gwC["USD"](100)), require(lines(alice,3)));
+        if (thirdLineCreatesLE)
+        {
+            env(trust(creator, assistor["USD"](100)), require(lines(creator, 3)));
+        }
+        else
+        {
+            env(trust(creator, assistor["USD"](100)),require(lines(creator,3)));
+
+            auto const lines = env.rpc ("json", "account_lines",
+                R"({"account": ")" + creator.human() + R"("})");
+            // Verify that all lines have 100 limit from creator
+            BEAST_EXPECT(lines[jss::result][jss::lines].isArray());
+            BEAST_EXPECT(lines[jss::result][jss::lines].size() == 3);
+            for (auto const & line : lines[jss::result][jss::lines])
+            {
+                BEAST_EXPECT(line[jss::limit] == "100");
+            }
+        }
     }
 
+    Json::Value trust_explicit_amt(jtx::Account const & a, STAmount const & amt)
+    {
+        Json::Value jv;
+        jv[jss::Account] = a.human();
+        jv[jss::LimitAmount] = amt.getJson(0);
+        jv[jss::TransactionType] = "TrustSet";
+        jv[jss::Flags] = 0;
+        return jv;
+    }
+
+    void testMalformedTransaction()
+    {
+        testcase("SetTrust checks for malformed transactions");
+
+        using namespace jtx;
+        Env env{ *this };
+
+        auto const gw = Account{ "gateway" };
+        auto const alice = Account{ "alice" };
+        env.fund(XRP(10000), gw, alice);
+
+        // Require valid tf flags
+        for (std::uint64_t badFlag = 1u ; badFlag <= std::numeric_limits<std::uint32_t>::max(); badFlag *= 2)
+        {
+            if( badFlag & tfTrustSetMask)
+                env(trust(alice, gw["USD"](100), static_cast<std::uint32_t>(badFlag)), ter(temINVALID_FLAG));
+        }
+        // trust amount can't be XRP
+        env(trust_explicit_amt(alice, drops(10000)), ter(temBAD_LIMIT));
+
+        // trust amount can't be badCurrency IOU
+        env(trust_explicit_amt(alice, gw[ to_string(badCurrency())](100)), ter(temBAD_CURRENCY));
+
+        // trust amount can't be negative
+        env(trust(alice, gw["USD"](-1000)), ter(temBAD_LIMIT));
+
+        // trust amount can't be from invalid issuer
+        env(trust_explicit_amt(alice, STAmount{Issue{to_currency("USD"), noAccount()}, 100 }), ter(temDST_NEEDED));
+
+        // trust cannot be to self
+        env(trust(alice, alice["USD"](100)), ter(temDST_IS_SRC));
+
+        // tfSetAuth flag should not be set if not required by account lsfRequireAuth
+        env(trust(alice, gw["USD"](100), tfSetfAuth), ter(tefNO_AUTH_REQUIRED));
+    }
+
+    void testModifyQualityOfTrustline(bool createQuality, bool createOnHighAcct)
+    {
+        using namespace std::string_literals;
+        testcase("SetTrust " + (createQuality ? "creates"s : "removes"s) + " quality of trustline for "
+        + (createOnHighAcct ? "high"s : "low"s ) +  " account" );
```

Comment:
I always forget about the `s` suffix and I confuse it with the `std::chrono` one for seconds - but I think it's fine to use, albeit consistently (some of the above have it, some don't). Or, perhaps simpler, just add it to the first fragment and then append unandorned `const char[]`s ? This is very nit-picky, for sure.